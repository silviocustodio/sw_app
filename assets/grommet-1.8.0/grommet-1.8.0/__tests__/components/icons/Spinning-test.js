// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Spinning from '../../../src/js/components/icons/Spinning';

describe('Spinning', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Spinning />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct small=true rendering', () => {
    const component = renderer.create(
      <Spinning small={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct rendering when size=small', () => {
    const component = renderer.create(
      <Spinning size="small" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct rendering when size=medium', () => {
    const component = renderer.create(
      <Spinning size="medium" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct rendering when size=large', () => {
    const component = renderer.create(
      <Spinning size="large" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct rendering when size=xlarge', () => {
    const component = renderer.create(
      <Spinning size="xlarge" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct rendering when size=huge', () => {
    const component = renderer.create(
      <Spinning size="huge" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct className rendering', () => {
    const component = renderer.create(
      <Spinning className='testing' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has microdata properties rendering', () => {
    const component = renderer.create(
      <Spinning itemScope={true} itemType="http://schema.org/Article"
        itemProp="test"/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
