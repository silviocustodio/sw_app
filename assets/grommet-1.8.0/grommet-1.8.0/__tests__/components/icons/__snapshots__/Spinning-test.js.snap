// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinning has correct className rendering 1`] = `
<svg
  aria-label="Spinning"
  className="grommetux-icon-spinning grommetux-icon-spinning--responsive testing"
  role="img"
  version="1.1"
  viewBox="0 0 48 48"
>
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="21"
    stroke="#979797"
    strokeWidth="6"
  />
</svg>
`;

exports[`Spinning has correct default options 1`] = `
<svg
  aria-label="Spinning"
  className="grommetux-icon-spinning grommetux-icon-spinning--responsive"
  role="img"
  version="1.1"
  viewBox="0 0 48 48"
>
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="21"
    stroke="#979797"
    strokeWidth="6"
  />
</svg>
`;

exports[`Spinning has correct rendering when size=huge 1`] = `
<svg
  aria-label="Spinning"
  className="grommetux-icon-spinning grommetux-icon-spinning--huge grommetux-icon-spinning--responsive"
  role="img"
  version="1.1"
  viewBox="0 0 48 48"
>
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="21"
    stroke="#979797"
    strokeWidth="6"
  />
</svg>
`;

exports[`Spinning has correct rendering when size=large 1`] = `
<svg
  aria-label="Spinning"
  className="grommetux-icon-spinning grommetux-icon-spinning--large grommetux-icon-spinning--responsive"
  role="img"
  version="1.1"
  viewBox="0 0 48 48"
>
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="21"
    stroke="#979797"
    strokeWidth="6"
  />
</svg>
`;

exports[`Spinning has correct rendering when size=medium 1`] = `
<svg
  aria-label="Spinning"
  className="grommetux-icon-spinning grommetux-icon-spinning--medium grommetux-icon-spinning--responsive"
  role="img"
  version="1.1"
  viewBox="0 0 48 48"
>
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="21"
    stroke="#979797"
    strokeWidth="6"
  />
</svg>
`;

exports[`Spinning has correct rendering when size=small 1`] = `
<svg
  aria-label="Spinning"
  className="grommetux-icon-spinning grommetux-icon-spinning--small grommetux-icon-spinning--responsive"
  role="img"
  version="1.1"
  viewBox="0 0 48 48"
>
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="21"
    stroke="#979797"
    strokeWidth="6"
  />
</svg>
`;

exports[`Spinning has correct rendering when size=xlarge 1`] = `
<svg
  aria-label="Spinning"
  className="grommetux-icon-spinning grommetux-icon-spinning--xlarge grommetux-icon-spinning--responsive"
  role="img"
  version="1.1"
  viewBox="0 0 48 48"
>
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="21"
    stroke="#979797"
    strokeWidth="6"
  />
</svg>
`;

exports[`Spinning has correct small=true rendering 1`] = `
<svg
  aria-label="Spinning"
  className="grommetux-icon-spinning grommetux-icon-spinning--small grommetux-icon-spinning--responsive"
  role="img"
  version="1.1"
  viewBox="0 0 48 48"
>
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="21"
    stroke="#979797"
    strokeWidth="6"
  />
</svg>
`;

exports[`Spinning has microdata properties rendering 1`] = `
<svg
  aria-label="Spinning"
  className="grommetux-icon-spinning grommetux-icon-spinning--responsive"
  itemProp="test"
  itemScope={true}
  itemType="http://schema.org/Article"
  role="img"
  version="1.1"
  viewBox="0 0 48 48"
>
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="21"
    stroke="#979797"
    strokeWidth="6"
  />
</svg>
`;
